pyomo/contrib/trustregion/TRF.py

```diff
@@ -1,45 +1,41 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-
 import numpy as np
 from math import pow
 from numpy import inf
 from numpy.linalg import norm
 from pyomo.contrib.trustregion.param import *
-from pyomo.contrib.trustregion.filterMethod import FilterElement, Filter
-from pyomo.contrib.trustregion.helper import cloneXYZ, packXYZ, minIgnoreNone, maxIgnoreNone
-from pyomo.contrib.trustregion.Logger import IterLog, Logger
-from pyomo.contrib.trustregion.PyomoInterface import PyomoInterface, ROMType
-
-
+from pyomo.contrib.trustregion.filterMethod import (
+    FilterElement, Filter)
+from pyomo.contrib.trustregion.helper import (cloneXYZ, packXYZ,
+    minIgnoreNone, maxIgnoreNone)
+from pyomo.contrib.trustregion.Logger import (IterLog, Logger)
+from pyomo.contrib.trustregion.PyomoInterface import (
+    PyomoInterface, ROMType)
 
 def TRF(m,eflist):
+    """The main function of the Trust Region Filter algorithm
+    m is a PyomoModel containing ExternalFunction() objects Model
+    requirements: m is a nonlinear program, with exactly one active
+    objective function.
+
+    eflist is a list of ExternalFunction objects that should be
+    treated with the trust region
+
+    Return: 
+    model is solved, variables are at optimal solution or
+    other exit condition.  model is left in reformulated form, with
+    some new variables introduced in a block named "tR" TODO: reverse
+    the transformation.
     """
-    The main function of the Trust Region Filter algorithm
-
-    m is a PyomoModel containing ExternalFunction() objects
-    Model requirements: m is a nonlinear program, with exactly one active objective function.
-
-    eflist is a list of ExternalFunction objects that should be treated with the
-    trust region
-
-
-    Return:
-    model is solved, variables are at optimal solution or other exit condition.
-    model is left in reformulated form, with some new variables introduced
-    in a block named "tR" TODO: reverse the transformation.
-    """
-
 
     logger = Logger()
     filteR = Filter()
-    problem = PyomoInterface(m,eflist)
+    problem = PyomoInterface(m, eflist)
     x, y, z = problem.getInitialValue()
 
+    trustRadius = CONFIG.trust_radius
```

Comment:
There is a slight misunderstanding of how to best use the ConfigBlock system.  The main `CONFIG` object should be thought of as a "template" that *defines* the acceptable options, and I usually store this as a static class member on the solver (or transformation) class.  Then, the solver in its `__init__()` creates an instance object by passing the `**kwds` sent to the solver.  This allows a user to specify the options as keyword arguments to the solver, without overriding the defaults written in by the solver developer (think about creating 2 copies of the same solver: by making a *copy* of the Config, both solvers start from the same default template).  For example:
```
class MySolver(plugin.Plugin):
    plugin.implements(IOptSolver)
    plugin.alias('mysolver')

    CONFIG = pyomo.config.trustregion.param.CONFIG
    # Actually, I would usually declare everything here -- to keep the related code 
    # together, but since you already have a param.py...

    def __init__(self, **kwds):
        super(MySolver, self).__init__(**kwds)
        self.config = MySolver.CONFIG(kwds)
```